### DISCUSSION TITLE
Brand New User - Trying to figure out how to get all row values into a Field Group

### DISCUSSION THREAD

**Discussion by mcancell** [Budibase/budibase#12887] [Help] [answered]
Hi,

New user here. I created a Data Provider, added a Options Picker, Added a Table, and finally added a Field Group. When I choose a value from the Picker, It does filter the table correctly and also DOES retrieve the value of the picked value into the corresponding field in the Field Group (e.g., Rate Name).  However, all of the other fields in the Group **never** get retrieved. 
The image below illustrated my problem. 
Why are all the other values NOT getting retrieved into the other Fields in the Group?
![Capture](https://github.com/Budibase/budibase/assets/25249301/42fee74f-a6f2-47c3-83c0-6c37c52d30d0)

**Comment by mcancell** [comment]
"Check 123, Check 123" - Just trying to see if anyone actually monitors these discussions.... ;-)

**Reply by andz-bb** [reply, marked as answer]
hey @mcancell, welcome to Budibase, we're here to help :)

The reason the Rate Name field below your table updates when you set the Option Picker value is that they're linked to the same value on the form. For the other fields in the Field Group, we need to set the value manually when the Option Picker value changes.

You can use [On Change actions](https://docs.budibase.com/docs/form-actions#on-change) on the Option Picker in order to update the other field values. The process will be the same for each field. Let's start with Application Developer.

<img width="1512" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/4f4af25e-2c8d-4aef-b9be-1bfe1c4ab5c5">

Select the Option Picker in  the component tree on the left side bar and then select **On change** from the component settings on the right side bar. A drawer will pop up from the bottom (as shown above) with different actions that can be triggered when the field value changes. The action we want for this is **Update Field Value**.

We can then select the relevant form and field. We want to set the value to a new value that we will calculate based on the selected value. I used the following JavaScript binding for this:

```js
const selectedRow = ($("Rates Data Provider.Rows").find(rate => rate._id === $("Field Value")))

return selectedRow ? selectedRow["Application Developer"] : null
```

This gets the full object for the selected rate, and then we return the Application Developer property for that specific object and use that to update the Application Developer field. Note that the value I'm checking is `_id` which will need to be what the Options picker uses as the Value column.

Hopefully this or similar will get that field to update each time you select a new rate at the top. You'll then need to repeat this process for the other fields. This is how the finished product will look:

![field group update demo](https://github.com/Budibase/budibase/assets/110921612/ee656377-38a1-40be-a04f-a8c2adeb3ff1)

If you have any questions at all, feel free to ask - although please note that it may take us time to get back to questions from the community :)

**Reply by mcancell** [reply]
Much thanks. I am going to try this shortly. I will let you know how it turns out. I think I did try doing the "Fetch Row", but I was doing that in the Field Group, not the Picker. And I dont think I used Javascript to retrieve the data. Let me give it a go.

**Reply by mcancell** [reply]
OMG! It works perfectly. Thank you so much. Now I can extraplate that process for my other forms. I just wanted to start with a very simple form. I am going to start a NEW discussion thread about something I am seeing re: data types.  I hope you can take a look at that too. Thx
